### DISCUSSION TITLE
Many buckets (multi-tenant application)

### DISCUSSION THREAD

**Discussion by AllanJard** [seaweedfs/seaweedfs#3988] [Q&A] [answered]
Hi,

I'm looking to use Seaweed to provide an S3 API for object storage for my multi-tenant application. I'd like to have one S3 bucket per tenant. I'm a little worried about the number of volumes that this would result in:

From the [S3 API document](https://github.com/seaweedfs/seaweedfs/wiki/Amazon-S3-API), it  says:

> Each bucket has its own collection. Usually one collection uses 7 volumes, where each volume is 30GB by default.

Based on that I've a few specific questions (note that my max bucket / collection size will be 2Gb):

1. Why 7 volumes? I don't see any reasoning for that in the docs.
2. Can that be configured - 1 volume per collection? Is there a downside to that.
3. Is a volume committed storage on the physical disc? Or can it be "oversold"?
4. This approach would result in thousands of buckets / collections. Is that okay?

Many thanks for any thoughts on this!

**Comment by chrislusf** [comment, marked as answer]
1. no particular reason.
2. yes. Less parallel writes.
3. can be oversold.
4. ok.

**Comment by AllanJard** [comment]
That's awesome - thank you so much :)
